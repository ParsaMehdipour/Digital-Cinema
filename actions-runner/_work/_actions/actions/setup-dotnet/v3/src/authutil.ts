import * as fs from 'fs';
import * as path from 'path';
import * as core from '@actions/core';
import * as github from '@actions/github';
import {XMLParser, XMLBuilder} from 'fast-xml-parser';

export function configAuthentication(
  feedUrl: string,
  existingFileLocation = '',
  processRoot: string = process.cwd()
) {
  const existingNuGetConfig: string = path.resolve(
    processRoot,
    existingFileLocation === ''
      ? getExistingNugetConfig(processRoot)
      : existingFileLocation
  );

  const tempNuGetConfig: string = path.resolve(
    processRoot,
    '../',
    'nuget.config'
  );

  writeFeedToFile(feedUrl, existingNuGetConfig, tempNuGetConfig);
}

function isValidKey(key: string): boolean {
  return /^[\w\-.]+$/i.test(key);
}

function getExistingNugetConfig(processRoot: string) {
  const defaultConfigName = 'nuget.config';
  const configFileNames = fs
    .readdirSync(processRoot)
    .filter(filename => filename.toLowerCase() === defaultConfigName);
  if (configFileNames.length) {
    return configFileNames[0];
  }
  return defaultConfigName;
}

function writeFeedToFile(
  feedUrl: string,
  existingFileLocation: string,
  tempFileLocation: string
) {
  core.info(
    `dotnet-auth: Finding any source references in ${existingFileLocation}, writing a new temporary configuration file with credentials to ${tempFileLocation}`
  );
  const sourceKeys: string[] = [];
  let owner: string = core.getInput('owner');
  const sourceUrl: string = feedUrl;
  if (!owner) {
    owner = github.context.repo.owner;
  }

  if (!process.env.NUGET_AUTH_TOKEN) {
    throw new Error(
      'The NUGET_AUTH_TOKEN environment variable was not provided. In this step, add the following: \r\nenv:\r\n  NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}'
    );
  }

  if (fs.existsSync(existingFileLocation)) {
    // get key from existing NuGet.config so NuGet/dotnet can match credentials
    const curContents: string = fs.readFileSync(existingFileLocation, 'utf8');

    const parserOptions = {
      ignoreAttributes: false
    };
    const parser = new XMLParser(parserOptions);
    const json = parser.parse(curContents);

    if (typeof json.configuration === 'undefined') {
      throw new Error(`The provided NuGet.config seems invalid.`);
    }
    if (json.configuration?.packageSources?.add) {
      const packageSources = json.configuration.packageSources.add;

      if (Array.isArray(packageSources)) {
        packageSources.forEach(source => {
          const value = source['@_value'];
          core.debug(`source '${value}'`);
          if (value.toLowerCase().includes(feedUrl.toLowerCase())) {
            const key = source['@_key'];
            sourceKeys.push(key);
            core.debug(`Found a URL with key ${key}`);
          }
        });
      } else {
        if (
          packageSources['@_value']
            .toLowerCase()
            .includes(feedUrl.toLowerCase())
        ) {
          const key = packageSources['@_key'];
          sourceKeys.push(key);
          core.debug(`Found a URL with key ${key}`);
        }
      }
    }
  }

  const xmlSource: any[] = [
    {
      '?xml': [
        {
          '#text': ''
        }
      ],
      ':@': {
        '@_version': '1.0'
      }
    },
    {
      configuration: [
        {
          config: [
            {
              add: [],
              ':@': {
                '@_key': 'defaultPushSource',
                '@_value': sourceUrl
              }
            }
          ]
        }
      ]
    }
  ];

  if (!sourceKeys.length) {
    const keystring = 'Source';

    xmlSource[1].configuration.push({
      packageSources: [
        {
          add: [],
          ':@': {
            '@_key': keystring,
            '@_value': sourceUrl
          }
        }
      ]
    });

    sourceKeys.push(keystring);
  }

  const packageSourceCredentials: any[] = [];
  sourceKeys.forEach(key => {
    if (!isValidKey(key)) {
      throw new Error(
        "Source name can contain letters, numbers, and '-', '_', '.' symbols only. Please, fix source name in NuGet.config and try again."
      );
    }

    packageSourceCredentials.push({
      [key]: [
        {
          add: [],
          ':@': {
            '@_key': 'Username',
            '@_value': owner
          }
        },
        {
          add: [],
          ':@': {
            '@_key': 'ClearTextPassword',
            '@_value': process.env.NUGET_AUTH_TOKEN
          }
        }
      ]
    });
  });

  xmlSource[1].configuration.push({
    packageSourceCredentials
  });

  const xmlBuilderOptions = {
    format: true,
    ignoreAttributes: false,
    preserveOrder: true,
    allowBooleanAttributes: true,
    suppressBooleanAttributes: true,
    suppressEmptyNode: true
  };

  const builder = new XMLBuilder(xmlBuilderOptions);

  const output = builder.build(xmlSource).trim();

  fs.writeFileSync(tempFileLocation, output);
}
