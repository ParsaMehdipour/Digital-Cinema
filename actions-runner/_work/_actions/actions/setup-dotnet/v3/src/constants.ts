/** NuGet lock file patterns */
export const lockFilePatterns = ['packages.lock.json'];

/**
 * .NET CLI command to list local NuGet resources.
 * @see https://docs.microsoft.com/dotnet/core/tools/dotnet-nuget-locals
 */
export const cliCommand =
  'dotnet nuget locals all --list --force-english-output';

export enum State {
  CachePrimaryKey = 'CACHE_KEY',
  CacheMatchedKey = 'CACHE_RESULT'
}

export enum Outputs {
  CacheHit = 'cache-hit',
  DotnetVersion = 'dotnet-version'
}
