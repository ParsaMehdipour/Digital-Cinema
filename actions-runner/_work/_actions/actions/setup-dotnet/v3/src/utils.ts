export const IS_WINDOWS = process.platform === 'win32';
export const IS_LINUX = process.platform === 'linux';
