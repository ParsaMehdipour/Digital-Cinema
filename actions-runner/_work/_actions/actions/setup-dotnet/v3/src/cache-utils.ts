import * as cache from '@actions/cache';
import * as core from '@actions/core';
import * as exec from '@actions/exec';

import {cliCommand} from './constants';

type NuGetFolderName =
  | 'http-cache'
  | 'global-packages'
  | 'temp'
  | 'plugins-cache';

/**
 * Get NuGet global packages, cache, and temp folders from .NET CLI.
 * @returns (Folder Name)-(Path) mappings
 * @see https://docs.microsoft.com/nuget/consume-packages/managing-the-global-packages-and-cache-folders
 * @example
 * Windows
 * ```json
 * {
 *   "http-cache": "C:\\Users\\user1\\AppData\\Local\\NuGet\\v3-cache",
 *   "global-packages": "C:\\Users\\user1\\.nuget\\packages\\",
 *   "temp": "C:\\Users\\user1\\AppData\\Local\\Temp\\NuGetScratch",
 *   "plugins-cache": "C:\\Users\\user1\\AppData\\Local\\NuGet\\plugins-cache"
 * }
 * ```
 *
 * Mac/Linux
 * ```json
 * {
 *   "http-cache": "/home/user1/.local/share/NuGet/v3-cache",
 *   "global-packages": "/home/user1/.nuget/packages/",
 *   "temp": "/tmp/NuGetScratch",
 *   "plugins-cache": "/home/user1/.local/share/NuGet/plugins-cache"
 * }
 * ```
 */
export const getNuGetFolderPath = async () => {
  const {stdout, stderr, exitCode} = await exec.getExecOutput(
    cliCommand,
    undefined,
    {ignoreReturnCode: true, silent: true}
  );

  if (exitCode) {
    throw new Error(
      !stderr.trim()
        ? `The '${cliCommand}' command failed with exit code: ${exitCode}`
        : stderr
    );
  }

  const result: Record<NuGetFolderName, string> = {
    'http-cache': '',
    'global-packages': '',
    temp: '',
    'plugins-cache': ''
  };

  const regex = /(?:^|\s)(?<key>[a-z-]+): (?<path>.+[/\\].+)$/gm;

  let match: RegExpExecArray | null;
  while ((match = regex.exec(stdout)) !== null) {
    const key = match.groups!.key;
    if ((key as NuGetFolderName) in result) {
      result[key] = match.groups!.path;
    }
  }

  return result;
};

export function isCacheFeatureAvailable(): boolean {
  if (cache.isFeatureAvailable()) {
    return true;
  }

  if (isGhes()) {
    core.warning(
      'Cache action is only supported on GHES version >= 3.5. If you are on version >=3.5 Please check with GHES admin if Actions cache service is enabled or not.'
    );
    return false;
  }
  core.warning(
    'The runner was not able to contact the cache service. Caching will be skipped'
  );

  return false;
}

/**
 * Returns this action runs on GitHub Enterprise Server or not.
 * (port from https://github.com/actions/toolkit/blob/457303960f03375db6f033e214b9f90d79c3fe5c/packages/cache/src/internal/cacheUtils.ts#L134)
 */
function isGhes(): boolean {
  const url = process.env['GITHUB_SERVER_URL'] || 'https://github.com';
  return new URL(url).hostname.toUpperCase() !== 'GITHUB.COM';
}
