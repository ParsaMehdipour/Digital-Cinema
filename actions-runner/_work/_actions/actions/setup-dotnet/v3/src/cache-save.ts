import * as core from '@actions/core';
import * as cache from '@actions/cache';
import fs from 'node:fs';
import {getNuGetFolderPath} from './cache-utils';
import {State} from './constants';

// Catch and log any unhandled exceptions.  These exceptions can leak out of the uploadChunk method in
// @actions/toolkit when a failed upload closes the file descriptor causing any in-process reads to
// throw an uncaught exception.  Instead of failing this action, just warn.
process.on('uncaughtException', e => {
  const warningPrefix = '[warning]';
  core.info(`${warningPrefix}${e.message}`);
});

export async function run() {
  try {
    if (core.getBooleanInput('cache')) {
      await cachePackages();
    }
  } catch (error) {
    core.setFailed(error.message);
  }
}

const cachePackages = async () => {
  const state = core.getState(State.CacheMatchedKey);
  const primaryKey = core.getState(State.CachePrimaryKey);

  if (!primaryKey) {
    core.info('Primary key was not generated, not saving cache.');
    return;
  }

  const {'global-packages': cachePath} = await getNuGetFolderPath();

  if (!fs.existsSync(cachePath)) {
    throw new Error(
      `Cache folder path is retrieved for .NET CLI but doesn't exist on disk: ${cachePath}`
    );
  }

  if (primaryKey === state) {
    core.info(
      `Cache hit occurred on the primary key ${primaryKey}, not saving cache.`
    );
    return;
  }

  const cacheId = await cache.saveCache([cachePath], primaryKey);
  if (cacheId == -1) {
    return;
  }

  core.info(`Cache saved with the key: ${primaryKey}`);
};

run();
