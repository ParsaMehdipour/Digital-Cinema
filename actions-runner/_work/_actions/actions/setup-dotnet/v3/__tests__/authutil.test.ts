import * as io from '@actions/io';
import fs from 'fs';
import path from 'path';

const fakeSourcesDirForTesting = path.join(
  __dirname,
  'runner',
  path.join(Math.random().toString(36).substring(7)),
  's'
);

const invalidNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>`;

const emptyNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
</configuration>`;

const nugetorgNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const gprnugetorgNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="GPR" value="https://nuget.pkg.github.com/OwnerName/index.json" protocolVersion="3" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const gprNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="GPR" value="https://nuget.pkg.github.com/OwnerName/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const twogprNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="GPR-GitHub" value="https://nuget.pkg.github.com/OwnerName/index.json" protocolVersion="3" />
    <add key="GPR-Actions" value="https://nuget.pkg.github.com/actions/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const spaceNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="GPR GitHub" value="https://nuget.pkg.github.com/OwnerName/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const azureartifactsNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="AzureArtifacts" value="https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

const azureartifactsnugetorgNuGetConfig = `<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="AzureArtifacts" value="https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json" protocolVersion="3" />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
</configuration>`;

// We want a NuGet.config one level above the sources directory, so it doesn't trample a user's NuGet.config but is still picked up by NuGet/dotnet.
const nugetConfigFile = path.join(fakeSourcesDirForTesting, '../nuget.config');

process.env['GITHUB_REPOSITORY'] = 'OwnerName/repo';
import * as auth from '../src/authutil';

describe('authutil tests', () => {
  beforeEach(async () => {
    await io.rmRF(fakeSourcesDirForTesting);
    await io.mkdirP(fakeSourcesDirForTesting);
  }, 30000);

  afterAll(async () => {
    await io.rmRF(fakeSourcesDirForTesting);
  }, 30000);

  beforeEach(() => {
    if (fs.existsSync(nugetConfigFile)) {
      fs.unlinkSync(nugetConfigFile);
    }
    process.env['INPUT_OWNER'] = '';
    process.env['NUGET_AUTH_TOKEN'] = '';
  });

  it('no existing config, sets up a full NuGet.config with URL and user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('no existing config, auth token environment variable not provided, throws', async () => {
    let thrown = false;
    try {
      auth.configAuthentication(
        'https://nuget.pkg.github.com/OwnerName/index.json',
        '',
        fakeSourcesDirForTesting
      );
    } catch {
      thrown = true;
    }
    expect(thrown).toBe(true);
  });

  it('no existing config, sets up a full NuGet.config with URL and other owner/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    process.env['INPUT_OWNER'] = 'otherorg';
    auth.configAuthentication(
      'https://nuget.pkg.github.com/otherorg/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config (invalid), tries to parse an invalid NuGet.config and throws', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, invalidNuGetConfig);
    let thrown = false;
    try {
      auth.configAuthentication(
        'https://nuget.pkg.github.com/OwnerName/index.json',
        '',
        fakeSourcesDirForTesting
      );
    } catch {
      thrown = true;
    }
    expect(thrown).toBe(true);
  });

  it('existing config w/ no sources, sets up a full NuGet.config with URL and user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, emptyNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ no GPR sources, sets up a full NuGet.config with URL and user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, nugetorgNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ only GPR source, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, gprNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ GPR source and NuGet.org, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, gprnugetorgNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ two GPR sources, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, twogprNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ spaces in key, throws for now', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, spaceNuGetConfig);
    let thrown = false;
    try {
      auth.configAuthentication(
        'https://nuget.pkg.github.com/OwnerName/index.json',
        '',
        fakeSourcesDirForTesting
      );
    } catch {
      thrown = true;
    }
    expect(thrown).toBe(true);
  });

  it('existing config not in repo root, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigDirectory: string = path.join(
      fakeSourcesDirForTesting,
      'subfolder'
    );
    const inputNuGetConfigPath: string = path.join(
      inputNuGetConfigDirectory,
      'nuget.config'
    );
    fs.mkdirSync(inputNuGetConfigDirectory, {recursive: true});
    fs.writeFileSync(inputNuGetConfigPath, gprNuGetConfig);
    auth.configAuthentication(
      'https://nuget.pkg.github.com/OwnerName/index.json',
      'subfolder/nuget.config',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ only Azure Artifacts source, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, azureartifactsNuGetConfig);
    auth.configAuthentication(
      'https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('existing config w/ Azure Artifacts source and NuGet.org, sets up a partial NuGet.config user/PAT for GPR', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    const inputNuGetConfigPath: string = path.join(
      fakeSourcesDirForTesting,
      'nuget.config'
    );
    fs.writeFileSync(inputNuGetConfigPath, azureartifactsnugetorgNuGetConfig);
    auth.configAuthentication(
      'https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });

  it('no existing config, sets up a full NuGet.config with URL and token for other source', async () => {
    process.env['NUGET_AUTH_TOKEN'] = 'TEST_FAKE_AUTH_TOKEN';
    auth.configAuthentication(
      'https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json',
      '',
      fakeSourcesDirForTesting
    );
    expect(fs.existsSync(nugetConfigFile)).toBe(true);
    expect(
      fs.readFileSync(nugetConfigFile, {encoding: 'utf8'})
    ).toMatchSnapshot();
  });
});
