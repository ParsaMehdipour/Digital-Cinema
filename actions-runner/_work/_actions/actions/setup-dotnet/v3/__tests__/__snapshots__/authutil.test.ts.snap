// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authutil tests existing config not in repo root, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSourceCredentials>
    <GPR>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </GPR>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ Azure Artifacts source and NuGet.org, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json\\"/>
  </config>
  <packageSourceCredentials>
    <AzureArtifacts>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </AzureArtifacts>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ GPR source and NuGet.org, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSourceCredentials>
    <GPR>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </GPR>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ no GPR sources, sets up a full NuGet.config with URL and user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSources>
    <add key=\\"Source\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </packageSources>
  <packageSourceCredentials>
    <Source>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </Source>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ no sources, sets up a full NuGet.config with URL and user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSources>
    <add key=\\"Source\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </packageSources>
  <packageSourceCredentials>
    <Source>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </Source>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ only Azure Artifacts source, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json\\"/>
  </config>
  <packageSourceCredentials>
    <AzureArtifacts>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </AzureArtifacts>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ only GPR source, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSourceCredentials>
    <GPR>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </GPR>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests existing config w/ two GPR sources, sets up a partial NuGet.config user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com\\"/>
  </config>
  <packageSourceCredentials>
    <GPR-GitHub>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </GPR-GitHub>
    <GPR-Actions>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </GPR-Actions>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests no existing config, sets up a full NuGet.config with URL and other owner/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/otherorg/index.json\\"/>
  </config>
  <packageSources>
    <add key=\\"Source\\" value=\\"https://nuget.pkg.github.com/otherorg/index.json\\"/>
  </packageSources>
  <packageSourceCredentials>
    <Source>
      <add key=\\"Username\\" value=\\"otherorg\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </Source>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests no existing config, sets up a full NuGet.config with URL and token for other source 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json\\"/>
  </config>
  <packageSources>
    <add key=\\"Source\\" value=\\"https://pkgs.dev.azure.com/amullans/_packaging/GitHubBuilds/nuget/v3/index.json\\"/>
  </packageSources>
  <packageSourceCredentials>
    <Source>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </Source>
  </packageSourceCredentials>
</configuration>"
`;

exports[`authutil tests no existing config, sets up a full NuGet.config with URL and user/PAT for GPR 1`] = `
"<?xml version=\\"1.0\\"?>
<configuration>
  <config>
    <add key=\\"defaultPushSource\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </config>
  <packageSources>
    <add key=\\"Source\\" value=\\"https://nuget.pkg.github.com/OwnerName/index.json\\"/>
  </packageSources>
  <packageSourceCredentials>
    <Source>
      <add key=\\"Username\\" value=\\"OwnerName\\"/>
      <add key=\\"ClearTextPassword\\" value=\\"TEST_FAKE_AUTH_TOKEN\\"/>
    </Source>
  </packageSourceCredentials>
</configuration>"
`;
